packages/react/src/components/Authenticator/SignUp/FormFields.tsx

```diff
@@ -16,12 +21,28 @@ export function FormFields() {
   ) as SignUpContext;
   const { loginMechanisms, signUpAttributes } = _state.context.config;
 
+  const [order, setOrder] = React.useState([]);
+
   const fieldNames = Array.from(
     new Set([...loginMechanisms, ...signUpAttributes])
   );
 
+  const formOverrides = getActorState(_state).context?.formFields?.signUp;
+
   // Only 1 is supported, so `['email', 'phone_number']` will only show `email`
-  const loginMechanism = fieldNames.shift() as LoginMechanism;
+  const loginMechanism = fieldNames.shift() as LoginMechanism | CommonFields;
+
+  const userOR = formOverrides?.[loginMechanism];
+  const common = [
+    loginMechanism,
+    'password',
+    'confirm_password',
+  ] as CommonFields[];
+
+  React.useEffect(() => {
+    const fieldNamesCombined = [...common, ...fieldNames];
+    setOrder(setFormOrder(formOverrides, fieldNamesCombined));
+  }, []);
```

Comment:
Perhaps in the future we can look at this. But the `signUp` page is getting slower. It re-renders 7 times for every keyboard event. This was even before my refactor. Since we are now setting the order, it was causing the page to be even slower. Wrapping this in `useEffect` helped.